### PR TITLE
Add Commit Status Verification to deploy-template

### PR DESCRIPTION
**Pull request by rjohnson2011**
## Summary

The workflow prevents deploying failed commits by modifying `gitHub.sha` to `github.event.workflow_run.head_commit.id`. See [thread](https://oddball.slack.com/archives/C045LCF8WU8/p1735237544597019) for more details.

PR output:

```
Echo PR before push:
Run echo “github.sha: 17f0c2bc1de44575714094f32492750596ed4e16”
github.sha: 17f0c2bc1de44575714094f32492750596ed4e16
github.event.push.head_commit.id:
github.event.pull_request.merge_commit_sha: 413c1608c7272b1bdaa1c46a38e30adcfff15a01
github.event.head_commit.id:
Echo PR code checks after push:
Run echo “github.sha: 781541c1b366fc5549896ae961bf9d6faeb40b75"
github.sha: 781541c1b366fc5549896ae961bf9d6faeb40b75
github.event.push.head_commit.id:
github.event.pull_request.merge_commit_sha:
github.event.head_commit.id: 781541c1b366fc5549896ae961bf9d6faeb40b75
Echo PR Build gha:
Run echo “github.sha: 07fa20cb5fff56f63f1f981a57ff14ec20d2ace0”
github.sha: 07fa20cb5fff56f63f1f981a57ff14ec20d2ace0
github.event.push.head_commit.id:
github.event.pull_request.merge_commit_sha:
github.event.head_commit.id:
github.event.workflow_run.head_commit.id: 781541c1b366fc5549896ae961bf9d6faeb40b75
```

## Related issue(s)

[Github Ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/96743)

### CODE REVIEW

File: .github/workflows/deploy-template.yml [hunk 1]
```diff
@@ -31,16 +31,11 @@ jobs:
...
         run: |
           echo "environments=${{ env.environments }}" >> $GITHUB_OUTPUT
-  release:
-    needs: [prepare-values]
-    if: github.ref == 'refs/heads/master'
-    runs-on: ubuntu-20.04
-    steps:
-      - uses: actions/checkout@v4
 
       - name: Configure AWS Credentials
```
Comment (rmtolmach) [inline .github/workflows/deploy-template.yml:40]: This step was in the `release` section, but now it's in the `prepare-values` section. Was that intentional or

File: .github/workflows/deploy-template.yml [hunk 2]
```diff
@@ -49,6 +44,28 @@ jobs:
...
+      - name: Install jq
+        run: sudo apt-get update && sudo apt-get install -y jq
+
+      - name: Verify commit status
+        run: |
+          TOKEN=$(aws ssm get-parameter --name /devops/VA_VSP_BOT_GITHUB_TOKEN --with-decryption --query Parameter.Value --output text)
+
+          STATUS=$(curl -s -H "Authorization: token $TOKEN" \
+            "https://api.github.com/repos/department-of-veterans-affairs/vets-api/commits/${{ inputs.commit_sha }}/status" \
+            | jq -r '.state')
```
Comment (rmtolmach) [inline .github/workflows/deploy-template.yml:56]: The only check we care about passing is Code Checks. When I was on support, I saw `Build And Publish Preview Environment` fail a lot, but that one doesn't matter to us.

**Comment by RachalCassity** [comment]
I found the error: [Build, Push, & Deploy](https://github.com/department-of-veterans-affairs/vets-api/actions/workflows/build.yml) action is rendering the incorrect image tag. I’m digging further. 
[@Ryan Johnson](https://oddball.slack.com/team/U01BS7YK9ML)
 The [${{ github.sha }}](https://github.com/department-of-veterans-affairs/vets-api/blob/master/.github/workflows/build.yml#L55) is incorrect. We need to fix the incorrect github.sha issue. This pull request will not fix the issue.

**Review by RachalCassity — CHANGES_REQUESTED** [review]
Please fix the github.sha issue

**Comment by rjohnson2011** [comment]
> Please fix the github.sha issue

Made the changes discussed to `build.yml` and `deploy-template.yml`

**Review by LindseySaari — APPROVED** [review]
@rjohnson2011 Please place the before/after output in the PR description but otherwise looks good!

**Pull request merged**